rmw_zenoh_cpp/src/detail/zenoh_router_check.cpp

```diff
@@ -43,7 +43,12 @@ rmw_ret_t zenoh_router_check(z_session_t session)
     };
 
   rmw_ret_t ret = RMW_RET_OK;
-  z_owned_closure_zid_t router_callback = z_closure(callback, nullptr /* drop */, &context);
+  // z_owned_closure_zid_t router_callback = z_closure(callback, nullptr /* drop */, &context);
+  z_owned_closure_zid_t router_callback;
```

Comment:
Could you explain this change?

Reply:
`z_closure` is a define which I believe use syntax that it's not compatible with the windows compiler

Reply:
I see. It would be good to document this. Also let's add a function `make_z_closure()` to the `anonymous namesapce` that constructs and returns the `z_owned_closure_zid_t`. That way we implement it once somewhere and we simply update the `z_closure()` calls everywhere to `make_z_closure()`.

Reply:
@ahcorde I think you may have missed this suggestion.

Reply:
There are different `z_owned_closure_*_t`, I can create a template function. Where is the right file to include this new function ?

Reply:
That would be great! I'd put the function in `detail/rmw_data_types.hpp` right above where the `sub_data_handler` is defined!